source/content/guides/composer-convert.md

```diff
@@ -1,156 +1,308 @@
 
-This will set up the Multidev environment to receive and demo our changed code.
+1. In your local terminal, change directories to the site project. For example, if you keep your projects in a folder called `projects` in the home directory:
 
-## Create a New Composer Project
+  ```bash{promptUser:user}
+  cd ~/projects/$SITE/
+  ```
 
-1. In your local terminal, from the repository root of your Pantheon site, move a directory up:
+1. Add the Pantheon Drupal Upstream as a new remote called `ic`, fetch the `ic` branch, and checkout to a new local branch based on it called `composerify`:
 
   ```bash{promptUser:user}
-  cd ..
+  git remote add ic git@github.com:pantheon-upstreams/drupal-project.git && git fetch ic && git checkout -b composerify ic/master
   ```
 
-2. Use Composer to create a new project, using the [Pantheon Drupal 8 Composer](https://github.com/pantheon-systems/example-drops-8-composer) repository:
+  If you prefer, you can replace `composerify` with another branch name. If you do, remember to adjust the other examples in this doc to match.
 
-    ```bash{promptUser:user}
-    composer create-project pantheon-systems/example-drops-8-composer $site-composer
-    cd $site-composer
-    ```
+1. Copy any existing configuration from the default branch. If no files are copied through this step, that's ok:
+
+  ```bash{promptUser:user}
+  git checkout master sites/default/config
+  git mv sites/default/config/* config
+  git rm -f sites/default/config/.htaccess
+  git commit -m "Pull in configuration from default branch"
+  ```
 
-This will create a new directory based on the example project [pantheon-systems/example-drops-8-composer](https://github.com/pantheon-systems/example-drops-8-composer) in the `$site-composer` directory.
+1. Check for `pantheon.yml` settings you need to preserve by comparing your old codebase's `pantheon.yml` to the new `pantheon.upstream.yml`:
 
-### Copy pantheon.upstream.yml
+  ```bash{promptUser:user}
+  git diff master:pantheon.yml pantheon.upstream.yml
+  ```
 
-Since the drops-8 upstream has a `pantheon.upstream.yml` and the example-drops-8-composer upstream does not, copy over our old file for the platform to properly load the site. From the `$site-composer` directory, run:
+  Press `q` on your keyboard to exit the diff display.
 
-```bash{promptUser:user}
-cp ../$site/pantheon.upstream.yml .
-```
+   - If there are settings from `pantheon.yml` (shown with a `-` in the diff output), consider copying over your old `pantheon.yml` to preserve these settings:
+
+     ```bash{promptUser:user}
+     git checkout master pantheon.yml
+     git add pantheon.yml
+     git commit -m 'Copy my pantheon.yml'
+     ```
 
-`ls` should reveal that the new code repository now has a copy of the `pantheon.upstream.yml`.
+   If you prefer to keep the value for `database` from `pantheon.upstream.yml`, remove it from `pantheon.yml`.
```

Comment:
:rotating_light: :rotating_light: :rotating_light:  DANGER :rotating_light: :rotating_light: :rotating_light:
If a user has an existing site that's running Mariadb 10.0 and you follow this guide and end up with a pantheon.yml or pantheon.upstream.yml with a database version of 10.4 before Drupal 9 General Availability, your site might be in danger of database corruption.

Reply:
Currently trying to this on an actual project. I set 
```yml
api_version: 1
database:
  version: 10.0
```

on my `pantheon.yml` to prevent using 10.4

but I got 

```
remote: 
remote: PANTHEON ERROR:
remote: 
remote: Changes to `pantheon.yml` detected, but there was an error while processing it:
remote: 
remote: pantheon.yml: 
remote: Validation failed with error:
remote: >   10.0 is not one of [10.4]
remote: 
remote: while validating the following value:
remote: >   10.0
remote: >   ...
remote: 
```

Reply:
@hanoii The `database:` directive in pantheon.yml is not ready to use, and is dangerous. You must remove the directive if you do not want to use it, because at the moment the only valid value is 10.4. If your pantheon.upstream.yml has the `database:` directive, though, then you must remove it there as well; failure to do so prior to GA could be dangerous. An individual site is not supposed to modify the pantheon.upstream.yml file, though, so you will have extra work to do later if you do this. I would like to reiterate that if you make a mistake during this process, the result could be dangerous. If you think that you may have applied a `database` directive with the value 10.4 in any existing site's multidev, I would recommend deleting that multidev, and start over with a fresh one.

The recommended process to follow today is https://github.com/pantheon-systems/documentation/pull/6314. I would follow this guide to convert an existing drops-8 to integrated composer, even if your intention is to downgrade back to Drupal 8.

Reply:
@greg-1-anderson thanks! So no `database:` directive on either `pantheon.yml` and `pantheon.upstream.yml` means `10.0`? Isn't there anyway for me to override the `database` on my `pantheon.yml` with null or empty object something that would mean the same as no database? 

Anyway, I will try removing it. Do you foresee any issue moving forward that makes me reconsider doing the d8 to a new d9 site worthwhile considering at this point? I assume that  changing the upstream yaml, and then reverting that commit should allow for proper upstream merging from now on once 10.4 is GA.

Reply:
There is a comment above from mbaynton that I forgot about: it appears that this you can get 10.0 by modifying only pantheon.yml as follows:
```
database:
  {}
```
My honest advice, though, is that you should follow the instructions to convert your site to Drupal 9 by creating a new site, and then move your work back to your existing site after D9 GA.

Reply:
@greg-1-anderson ok. If I move it to a new d9 site I guess I can simply move everything there and then switch domains. Will explore this option. Thanks!

Reply:
FWIW, `database: {}` does work in that I get the push to go through and the site/code looks to have built, howeveer I connected to the mysql site and it reported to be `10.4`. The site worked, but I am not trying without any database at all.

 I did find a few things odd. The `tag pantheon_composerify_1` actually has the code from `master` (not yet composerified), but the deployed code as seen by SFTP looks the new stuff. This seems to be some odd issue with multidev and IC. I also saw the same behavior on a new d9 site I am using for testing where I created a new multidev. There it is less obvious because the tag belongs to the last commit, but there seems to be no artifacts tag on multidevs.

Reply:
Here's mine from my last project:
```yaml
api_version: 1
web_docroot: true
php_version: 7.4
drush_version: 10
database:
  version: 10.4
```

Reply:
FYI: Decoupled react with D9 backend: https://live-freshdrupalmi.pantheonsite.io